Dockerfile

```
@@ -207,6 +234,82 @@ RUN mkdir src &&\
     cmake \
         --build src/build \
         --target install \
+	-j$NPROC \
     &&\
     ln -s ${__prefix}/bin/thisdd4hep.sh ${__ldmx_env_script_d__}/thisdd4hep.sh &&\
     rm -r src
+
+###############################################################################
+# LHAPDF
+#
+# Needed for GENIE
+#
+###############################################################################
+LABEL lhapdf.version="6.5.2"
+ENV PYTHON_VERSION=3
+RUN mkdir src &&\
+    ${__wget} https://lhapdf.hepforge.org/downloads/?f=LHAPDF-6.5.2.tar.gz |\
+     ${__untar} &&\
+    cd src &&\
+    ./configure --prefix=${__prefix} &&\
+    make -j$NPROC install &&\
+    cd ../ &&\
+    rm -rf src
+
+###############################################################################
+# Installing log4cpp
+#
+# Needed for GENIE
+#
+###############################################################################
+RUN mkdir src &&\
+    ${__wget} https://sourceforge.net/projects/log4cpp/files/latest/download |\
+    ${__untar} &&\
+    cd src/log4cpp &&\
+    ./autogen.sh &&\
+    ./configure &&\
+    make check &&\
+    make install -j$NPROC &&\
+    cd ../.. &&\
+    rm -r src
+
+###############################################################################
+# GENIE
+#
+# Needed for ... GENIE :)
+#
+###############################################################################
+LABEL genie.version=3.02.00
+ENV GENIE_VERSION=3_02_00
+ENV GENIE_REWEIGHT_VERSION=1_02_00
+
+ENV GENIE=/usr/local/GENIE/Generator
+
+RUN mkdir -p /usr/local/root &&\
+    ln -s /usr/local/lib/root /usr/local/root/lib &&\
+    export ROOTSYS=/usr/local/root &&\
+    mkdir -p /usr/local/GENIE/Generator &&\
+    cd /usr/local/GENIE &&\
+    wget -q -O - https://github.com/GENIE-MC/Generator/archive/refs/tags/R-${GENIE_VERSION}.tar.gz | tar -xz -C Generator --strip-components 1 &&\
```

This looks good, I'm just putting the `tar` on a separate line
```suggestion
    wget -q -O - https://github.com/GENIE-MC/Generator/archive/refs/tags/R-${GENIE_VERSION}.tar.gz |\
      tar -xz -C Generator --strip-components 1 &&\
```

---

Done.